### DISCUSSION TITLE
xstate visualizer  not showing up when i run npm run build in react

### DISCUSSION THREAD

**Discussion by naveenkumardesai1** [statelyai/xstate#3701] [Q&A]
I am building xstate visualizer of state machine for my project using React
It works fine when I use npm start.

But when I am deploying to prod using npm run build, xstate visualizer not coming up, it shows blank.

Any idea why it works with npm start and not with npm run build

**Comment by davidkpiano** [comment]
Can you share code?

**Reply by naveenkumardesai1** [reply]
Thanks for response.
Here is my snippet
```
import React, { useEffect} from "react"
import { inspect } from "@xstate/inspect"
import { useMachine } from "@xstate/react"
import { createMachine } from "xstate"

export const StateMachine = () => {
    const toggleMachine = createMachine({
        id: "toggle",
        initial: "inactive",
        states: {
            inactive: { on: { TOGGLE: "active" } },
            active: { on: { TOGGLE: "inactive" } }
        }
    })

    const [, send, service] = useMachine(toggleMachine, {
        devTools: true
    })

    useEffect(() => {
        inspect({
            url: "https://stately.ai/viz/embed?zoom=1&panel=full&showOriginalLink=0&mode=viz&pan=1&controls=1&inspect"
        })
    }, [service, send])

    return (
        <>
            <div className="h-100 embed-responsive embed-responsive-21by9">
                <iframe title={"state-machine"} data-xstate="" className="d-flex" />
            </div>
        </>
    )
}
```
and my package.json, looks like
```
    "@xstate/inspect": "^0.4.1",
    "@xstate/react": "^3.0.0",
    "axios": "^0.27.2",
    "classnames": "^2.3.1",
    "dotenv": "^16.0.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-redux": "^8.0.2",
    "react-router-dom": "^6.3.0",
    "react-scripts": "5.0.1",
    "react-use": "^17.4.0",
    "redux": "^4.2.0",
    "sass": "^1.54.0",
    "typesafe-actions": "^4.4.2",
    "typescript": "^4.1.3",
    "web-vitals": "^2.1.4",
    "webpack-bundle-analyzer": "^4.6.1",
    "xstate": "^4.32.1"
```

Please note, ```npm start``` works, when we try to build in our production environment using ```npm run build```, not able to load the xstate diagram.

Any help would be great.

Thanks,
Naveen

**Reply by naveenkumardesai1** [reply]
Any idea on this, please?

**Reply by Andarist** [reply]
Please provide a runnable repro case that won't make us copy the code around. A GitHub repository to clone or a CodeSandbox with everything included would be very much appreciated.

**Reply by naveenkumardesai1** [reply]
Thanks for your response @Andarist 
I have created the repo and please check out and follow the instruction as in read me file

Repo:
https://github.com/naveenkumardesai1/state-machine-npm-build

Please let me know the issue at the earliest.

Thanks,
Naveen

**Reply by naveenkumardesai1** [reply]
Any help would be appreciated, please

**Reply by naveenkumardesai1** [reply]
any idea here, please?

**Reply by PiotrGrobelak** [reply]
Try to use the method inspect in your App file w/o useEffect. Example: https://codesandbox.io/s/xstate-react-viz-template-5wq3q

**Reply by naveenkumardesai1** [reply]
Thanks it works, but it opens new url (new tab) all together.
We are looking at something, which we embed together in iframe and in same page

We are writing something like
```
 inspect({
            url: "https://stately.ai/viz/embed?zoom=1&panel=full&showOriginalLink=0&mode=viz&pan=1&controls=1&inspect",
        })
    }
```
and we have iframe with ```data-xstate```
```
<iframe title={"state-machine"} data-xstate className="d-flex" />*/}
```

it does not render using npm run build but works with npm start
Any idea?

**Reply by naveenkumardesai1** [reply]
All good, it works as expected. Thanks
